cmd/daytona/main.go

```diff
@@ -181,6 +181,10 @@ func main() {
 	}
 
 	vaultConfig := api.DefaultConfig()
+	// Set the MaxRetries for rate-limited requests to a higher default
+	if vaultConfig.MaxRetries == 2 {
+		vaultConfig.MaxRetries = 5
```

Comment:
We should probably just set a value instead of checking if it's 2

Reply:
I don't think it's safe to assume that every user wants a max retry of 5. Why not just instruct users to leverage the `VAULT_RATE_LIMIT` env var?

Reply:
2 is a very low default and I was able to exhaust it easily.  Let's make the experience better for our users with a higher default.

Reply:
Also Daytona doesn't utilize the `config.ReadEnvironment()` method to override the default config.
https://godoc.org/github.com/hashicorp/vault/api#Config.ReadEnvironment

Reply:
`DefaultConfig` calls  `ReadEnvironment()`

I'm okay with defaulting to 5, but we shouldn't break the ability for users to leverage the expected sdk environment variables.

I'd suggest calling `ReadEnvironment()` before creating the client and we should be good.

Reply:
So this is interesting, `DefaultConfig` calls `ReadEnvironment`, but then overwrites `config.MaxRetries = 2` anyways. 🙃 
https://github.com/hashicorp/vault/blob/master/api/client.go#L164-L182

I can add the call ReadEnvironment again so that the `VAULT_MAX_RETRIES` works

Reply:
Done